src/Testing/TestCaseSourceLocatorFactory.php

```diff
@@ -52,10 +73,11 @@ public function create(): SourceLocator
 				if ($composerSourceLocator === null) {
 					continue;
 				}
-				$locators[] = $composerSourceLocator;
+				self::$composerSourceLocatorsCache[$cacheKey][] = $composerSourceLocator;
```

Comment:
One tiny detail - please first create a local variable and only after the foreach is done, assign it to the static property.

With the current approach if the 3rd iteration fails, the property still gets populated but next time it will just return two locators.